NickvisionTagger.Shared/Models/MusicFile.cs

```diff
@@ -533,6 +541,12 @@ public bool SaveTagToDisk(bool preserveModificationTimestamp)
             if (System.IO.Path.GetFileName(Path) != Filename)
             {
                 var newPath = $"{System.IO.Path.GetDirectoryName(Path)}{System.IO.Path.DirectorySeparatorChar}{Filename}";
+                Directory.CreateDirectory(System.IO.Path.GetDirectoryName(newPath)!);
+                var i = 1;
+                while (File.Exists(newPath))
+                {
+                    newPath = newPath.Remove(newPath.IndexOf($" ({i - 1})")) + $" ({i})";
```

Comment:
I wonder whether this will remove file extension.

Reply:
Good catch, it definitely would have. 

Fixed in latest commit.

Reply:
Ah but then I need to fix adding on the extension...please hold

Reply:
And I never did `i++`...

all fixed in latest commit.